### ISSUE TITLE
Rule 'directive-scope' for explicit scopes

### ISSUE THREAD

**Issue by tilmanpotthof** [Gillespie59/eslint-plugin-angular#255]
Add a rule similar to 'directive-restrict'. 

     {"restrict": "AE", "explicit": "never"}

'directive-scope'

    {"scope":["child", "isolate"], "explicit": "always"}

**Comment by apb7** [comment]
Hi @tilmanpotthof, is there any progress on this rule? Thanks!

**Comment by xdjinnx** [comment]
This repository is not that well maintained anymore. I'm guessing that he's not working on an implementation for this.
The owner of the repository seems to do some cleaning and merging of PR:s every month.

I'm guessing you or anyone else are welcome to implement the rule if you need it.
I might have some time to spare someday to fix this, but I would not count on it.

Also I don't really understand the rule,
Should the rule enforce you to declare a scope on the directive if "explicit": "always"?

Example:
```
angular.
module("animal").
directive('cat', function () {
    return {
        scope: true,
    };
});
```
or
```
angular.
module("animal").
directive('cat', function () {
    return {
        scope: {},
    };
});
```

**Comment by apb7** [comment]
Thanks for the reply @xdjinnx! Regarding "explicit": "always"  option, I think both of the above examples should be correct, that is, it should report an error only if no scope is declared. Thanks!
